packages/hardhat/contracts/strategy/CrabOTC.sol

```diff
@@ -0,0 +1,207 @@
+
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import {ICrabStrategyV2} from "../interfaces/ICrabStrategyV2.sol";
+import {IWETH9} from "../interfaces/IWETH9.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {EIP712} from "@openzeppelin/contracts/drafts/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/cryptography/ECDSA.sol";
+import {StrategyMath} from "./base/StrategyMath.sol";
+
+contract CrabOTC is ReentrancyGuard, EIP712 {
+    using StrategyMath for uint256;
+    using Address for address payable;
+
+    address public immutable crab;
+    address public immutable weth;
+    address public immutable controller;
+    address public immutable wPowerPerp;
+    mapping(address => mapping(uint256 => bool)) public nonces;
+
+    struct Order {
+        address initiator;
+        address trader;
+        uint256 quantity;
+        uint256 price;
+        bool isBuying;
+        uint256 expiry;
+        uint256 nonce;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    event DepositOTC(
+        address indexed depositor,
+        uint256 crabAmount,
+        uint256 wSqueethAmount,
+        uint256 depositedAmount,
+        uint256 executedPrice,
+        address trader
+    );
+
+    event WithdrawOTC(
+        address indexed depositor,
+        uint256 crabAmount,
+        uint256 ethSent,
+        uint256 wSqueethAmount,
+        uint256 executedPrice,
+        address trader
+    );
+
+    bytes32 private constant _CRAB_BALANCE_TYPEHASH =
+        keccak256("Order(address initiator,address trader,uint256 quantity,uint256 price,bool isBuying,uint256 expiry,uint256 nonce)");
+
+    constructor(address _crab) EIP712("CrabOTC", "2") {
+        require(_crab != address(0), "Invalid crab address");
+
+        crab = _crab;
+        weth = ICrabStrategyV2(_crab).weth();
+        controller = ICrabStrategyV2(_crab).powerTokenController();
+        wPowerPerp = ICrabStrategyV2(_crab).wPowerPerp();
+    }
+
+    /**
+     * @notice set nonce to true
+     * @param _nonce the number to be set true
+     */
+    function setNonceTrue(uint256 _nonce) external {
+        nonces[msg.sender][_nonce] = true;
+    }
+
+    function deposit(
+        uint256 _totalEth,
+        uint256 _minPrice,
+        Order memory _order
+    ) external payable nonReentrant {
+        _verifyOrder(_order, true, _minPrice);
+        uint256 depositedEth = msg.value;
+
+        uint256 wSqueethQuantity = _getWSqueethToMint(_totalEth);
+        require(_order.quantity >= wSqueethQuantity, "Order quantity is less than needed");
+
+        _order.quantity = wSqueethQuantity;
+
+        uint256 neededEth = _totalEth.sub(depositedEth);
+        uint256 wethAmount = _order.quantity.wmul(_order.price);
+        require(wethAmount >= neededEth, "Need more ETH");
```

Comment:
I'm not sure why we are declaring the `neededEth` variable, and what this require statement is for?

Reply:
You mean this one is checking that the amount of ETH will be gained from selling oSQTH is greater or equal to neededEth.